Storage/SQL/SQLiteBookmarksBase.swift

```diff
@@ -55,15 +55,18 @@ open class SQLiteBookmarks: BookmarksModelFactorySource, KeywordSearchSource {
         let sql = "SELECT bmkUri FROM view_bookmarksBuffer_on_mirror WHERE keyword = ? LIMIT 1"
         let args: Args = [keyword]
 
-        return self.db.runQueryConcurrently(sql, args: args, factory: { $0["bmkUri"] as! String })
+        let deferred = DeferredDBOperation<Maybe<String>>()
+
+        self.db.runQueryConcurrently(sql, args: args, factory: { $0["bmkUri"] as? String })
             >>== { cursor in
-                if cursor.status == .success {
-                    if let str = cursor[0] {
-                        return deferMaybe(str)
-                    }
-                }
+            guard cursor.status == .success, let str = cursor[0] as? String else {
```

Comment:
don't you need to turn the cursor into an array via `asArray` before accessing the array? Doesn't look like subscript(index: Int) is implemented. Or is the type here something else?

Reply:
https://github.com/mozilla-mobile/firefox-ios/blob/master/Storage/Cursor.swift#L47-L50